Tests/NIOSSHTests/EndToEndTests.swift

```diff
@@ -435,6 +771,121 @@ class EndToEndTests: XCTestCase {
         XCTAssertEqual(self.channel.activeServerChannels.count, 1)
         #endif
     }
+    
+    func testCustomPublicKeyAlgorithms() throws {
+        CustomPublicKey.wasUsed = false
+        NIOSSHAlgoritms.register(publicKey: CustomPublicKey.self, signature: CustomSignature.self)
```

Comment:
This kind of global state is really awkward for testing purposes because it makes it hard to remove. Can we consider registering things on a specific instance of configuration, rather than globally? That will help solve our various locking problems as well as our test issues.

Reply:
Do you mean in SSHClientConfiguration or SSHServerConfiguration?

Reply:
Sadly both, but we can use a single type stored in both places.

Reply:
Yeah, cool! On the same page then